account_statement_import_camt/models/parser.py

```diff
@@ -13,25 +13,73 @@ class CamtParser(models.AbstractModel):
     _name = "account.statement.import.camt.parser"
     _description = "Account Bank Statement Import CAMT parser"
 
-    def parse_amount(self, ns, node):
+    def parse_amount(self, ns, node, currency=None):
         """Parse element that contains Amount and CreditDebitIndicator."""
         if node is None:
-            return 0.0
+            return (None, None, None)
         sign = 1
         amount = 0.0
         sign_node = node.xpath("ns:CdtDbtInd", namespaces={"ns": ns})
         if not sign_node:
             sign_node = node.xpath("../../ns:CdtDbtInd", namespaces={"ns": ns})
         if sign_node and sign_node[0].text == "DBIT":
             sign = -1
+
         amount_node = node.xpath("ns:Amt", namespaces={"ns": ns})
-        if not amount_node:
+        if (
+            not amount_node
+            or currency is not None
+            and amount_node[0].xpath("./@Ccy", namespaces={"ns": ns})[0] != currency
+        ):
+            amount_node = node.xpath(
+                "./ns:AmtDtls/ns:CntrValAmt/ns:Amt", namespaces={"ns": ns}
+            )
+        if (
+            not amount_node
+            or currency is not None
+            and amount_node[0].xpath("./@Ccy", namespaces={"ns": ns})[0] != currency
+        ):
             amount_node = node.xpath(
                 "./ns:AmtDtls/ns:TxAmt/ns:Amt", namespaces={"ns": ns}
             )
-        if amount_node:
-            amount = sign * float(amount_node[0].text)
-        return amount
+        if (
+            not amount_node
+            or currency is not None
+            and amount_node[0].xpath("./@Ccy", namespaces={"ns": ns})[0] != currency
```

Comment:
Looks like you may be missing a couple of parenthesis here. Also, you could condense the code (making it more readable) by actually iterating over the various paths and breaking if the conditions are met.
```
for path in [ ... ]
    if
        break
else
    return (None, None, None))
```

I would feel safer if you'd raise in the situation that you *did* encounter one or more amount nodes but none of them were in the right currency.

Reply:
done